dapp_runner/runner/runner.py

```diff
@@ -186,7 +216,13 @@ async def stop(self):
             for s in cluster.instances:
                 service_tasks.extend(s._tasks)
 
+        networks = self._networks.values()
+        await asyncio.gather(*[n.remove() for n in networks])
+        proxies = self._proxies.values()
```

Comment:
I think we should stop the proxies before we remove the networks... 

or - even more importantly, I think the shutdown order should be:

1. shut down the proxies
2. shut down the clusters
3. shut down the networks 
... 
4. golem.stop, et eal
... ? 
5. profit ! 

;)

Reply:
I see, that makes sense, though shouldn't the clusters be the first to go still? Correct me if I'm wrong, but I see the clusters as the "application layer", while the proxies and networks are their infrastructure.

Reply:
hmm... well, while I'd agree that "networks" are infrastructure for the services, proxies _use_ the services ...

I believe I get your point... if you turn things around, it may be seen that from a perspective of a "service" as a product that the "proxy" is a piece of infrastructure ... but then, you could think that - in order to provide a single service, you need a network, a cluster running a parametrized runtime on providers and a local proxy -> from this POV, the network, the cluster (or, actually, the runtime) and the proxy are all part of infrastructure for the "service"...

and the runtime itself doesn't use the proxy...

Reply:
and answering your specific question -> if the clusters go first, then the proxies are immediately unusable... so, for the sake of gracefulness of the shutdown, the proxies (that work on top of the clusters) should go first, then the clusters (which work on top of the networks), then the networks...

Reply:
Ok, got it, thanks for explaining. :)

Reply:
Changed in 047bf10